p2p/dialer.go

```diff
@@ -169,11 +185,22 @@ func (m *dialer) exchangeConfig(consumerID, providerID identity.Identity, servic
 	if err != nil {
 		return nil, fmt.Errorf("could not pack signed message: %v", err)
 	}
-	_, err = m.sendSignedMsg(brokerConn, configExchangeACKSubject(providerID, serviceType), packedMsg, timeout)
+	_, err = m.sendSignedMsg(brokerConn, configExchangeACKSubject(providerID, serviceType), packedMsg, ctx)
 	if err != nil {
 		return nil, fmt.Errorf("could not send signed msg: %v", err)
 	}
 
+	// Put this under feature set check when its available
+	// Wait until provider confirms that channel handlers are ready
+	select {
+	case <-ready:
+		log.Debug().Msg("Received handlers ready message from provider")
+	case <-ctx.Done():
+		log.Debug().Msg("Timeout while performing configuration exchange - trying to continue")
+	case <-time.After(5 * time.Second):
```

Comment:
I would go simpler and return error if context is canceled, it will be strange to continue if you didn't managed to exchange in given time interval.

```go
select {
    case <-ready:
       return &p2pConnectConfig{....}
   case <-ctx.Done():
       return errors.New("timeout waiting for p2p channel config exchange")
}
```
```

Reply:
Do you suggest reducing initial Dial timeout? I've added separate for backwards compatibility. If provider does not support this message, not to wait for another 10sec.